Shared/Samples/List contents of KML file/ListContentsOfKMLFileView.swift

```diff
@@ -0,0 +1,261 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct ListContentsOfKMLFileView: View {
+    /// The view model for the sample.
+    @StateObject private var model = Model()
+    
+    var body: some View {
+        VStack(spacing: 0) {
+            Text("Tap on a disclosure to reveal a node's children. Tap on a node to open it in a scene.")
+                .multilineTextAlignment(.center)
+                .frame(maxWidth: .infinity, alignment: .center)
+                .padding(8)
+                .background(Color(.systemGroupedBackground))
+            
+            // Recursively displays the dataset's nodes in a list.
+            List(model.kmlDataset?.rootNodes ?? [], id: \.name, children: \.children) { node in
+                VStack(alignment: .leading) {
+                    NavigationLink {
+                        if let viewpoint = model.nodeViewpoints[node.name] {
+                            SceneView(scene: model.scene, viewpoint: viewpoint)
+                                .navigationTitle(node.name)
+                        } else {
+                            Text("This node has no extent to view.")
+                                .navigationTitle(node.name)
+                        }
+                    } label: {
+                        VStack(alignment: .leading) {
+                            if !node.name.isEmpty {
+                                Text(node.name)
+                            }
+                            Text(node.typeLabel)
+                                .font(.footnote)
+                        }
+#if targetEnvironment(macCatalyst)
+                        .padding(.leading)
+#endif
+                    }
+                }
+            }
+        }
+        .errorAlert(presentingError: $model.error)
+    }
+}
+
+// MARK: Model
+
+private extension ListContentsOfKMLFileView {
+    /// The view model for the sample.
+    @MainActor
+    final class Model: ObservableObject {
+        /// A dataset containing the KML data from a local file.
+        @Published private(set) var kmlDataset: KMLDataset?
+        
+        /// The viewpoints for the nodes in the dataset.
+        @Published private(set) var nodeViewpoints: [String: Viewpoint] = [:]
+        
+        /// The error shown in the error alert.
+        @Published var error: Error?
+        
+        /// A scene for displaying the KML data.
+        let scene: ArcGIS.Scene = {
+            let scene = Scene(basemapStyle: .arcGISImagery)
+            let elevationSource = ArcGISTiledElevationSource(url: .worldElevationService)
+            scene.baseSurface.addElevationSource(elevationSource)
+            return scene
+        }()
+        
+        /// The task used for the asynchronous setup operations.
+        private var setupTask: Task<Void, Error>?
```

Comment:
Does this need to be optional since we create it at init and never set it to `nil` ?

And the task can be ` Task<Void, Never>` since an error is never thrown in the task.

Reply:
I believe the task does have to be optional since `self` needs to be initialized before it can be used in the closure.